src/payment/payment.controller.ts

```diff
@@ -0,0 +1,53 @@
+import { Controller, Post, Body, UseGuards, BadRequestException } from '@nestjs/common';
+import { PaymentService } from './payment.service';
+import { CreatePaymentDto, PaymentOption } from './dto/create-payment.dto';
+import { UserGuard } from '../common/guards/user.guard';
+import { ApiBearerAuth, ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { CurrentUser } from '../common/decorators/current-user.decorator';
+
+type PaymentHandler = (
+  userId: string,
+  totalAmount: number,
+  paymentOption?: string
+) => Promise<void>;
+
+@Controller('payment')
+@UseGuards(UserGuard)
+@ApiTags('payment')
+@ApiBearerAuth()
+export class PaymentController {
+  private paymentHandlers: Record<string, PaymentHandler>;
+
+  constructor(private readonly paymentService: PaymentService) {
+    this.paymentHandlers = {
+      cash: async (userId, totalAmount, paymentOption) =>
+        this.paymentService.processCashPayment(
+          userId,
+          totalAmount,
+          paymentOption || PaymentOption.cash
+        ),
+      card: async (userId, totalAmount, paymentOption) =>
+        this.paymentService.processPayment(
+          userId,
+          totalAmount,
+          paymentOption || PaymentOption.card
+        ),
+    };
+  }
+
+  @ApiOperation({ summary: 'Handle payment' })
+  @ApiResponse({ status: 200, description: 'Payment processed successfully' })
+  @ApiResponse({ status: 400, description: 'Bad Request' })
+  @Post('')
+  async handlePayment(
+    @Body() createPaymentDto: CreatePaymentDto,
+    @CurrentUser('id') userId: string
+  ) {
+    const { totalAmount, paymentOption } = createPaymentDto;
```

Comment:
I don't think totalAmount should be received from body

Reply:
yep, looks like we need check such information from db so we send productId ( for our application is appointmentId) and amount (for our application is amount of days) and we calculate the sum